llvm_util/llvm2alive.cpp

```diff
@@ -818,10 +838,9 @@ class llvm2alive_ : public llvm::InstVisitor<llvm2alive_, unique_ptr<Instr>> {
     return true;
   }
 
-  optional<ParamAttrs> handleAttributes(llvm::Argument &arg) {
+  optional<ParamAttrs> handleAttributes(const llvm::AttributeSet &aset) {
     ParamAttrs attrs;
-    for (auto &attr : arg.getParent()->getAttributes()
-                         .getParamAttributes(arg.getArgNo())) {
+    for (auto &attr : aset) {
```

Comment:
why this change?  Sorry, I'm a bit confused with the attributes. They can come from either the call instruction or from the declaration, but I think there's an API that gives the union of those?

Reply:
The change was made to make the comment that suggests a change to use handleAttribute() (at line 281) consistent with the implementation.
For the API that returns the union of those, I also think the API should be at somewhere because it would be pretty convenient, but I couldn't find it. :/

Reply:
see here: https://llvm.org/doxygen/Instructions_8cpp_source.html#l00321
`paramHasAttr(argno, attr`

Reply:
That was helpful for removing the big if statement at visitCallInst, thanks.
However I feel I'm still not at the best answer; the main obstacle was that dereferenceable bytes should have retrieved case by case (it is possible that there exists such functionality but again I couldn't find it again :/).
Anyway, I feel if we fully support readonly/readnone/etc attached at call sites, things should become simpler. But I think it is not in high priority..

Reply:
Or, to minimize diff, I can just remove this change (the change at the argument of hnadleAttributes). What do you think?